include/boost/url/detail/config.hpp

```diff
@@ -84,4 +85,14 @@
 # error Unknown or unsupported architecture, please open an issue
 #endif
 
+namespace boost {
+
+namespace urls {
+
+using off_t = ::uint16_t; // private
```

Comment:
what's wrong with unsigned short?

Reply:
I was hoping you would comment on that. I was going to use `size_t` or `uint32_t` directly on the table class, then I got this pattern from http_proto. What I don't get is why `uint16_t` is OK for fields. Why is it?

Reply:
don't you think 64kb is enough for an HTTP header?

Reply:
For the median HTTP header, yes. For the worst-case HTTP header according to the spec, I don't know.

For the median URL header, yes. For the worst-case URL according to the spec, no because of a [technicality](https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers).